src/SignatureParameters.php

```diff
@@ -31,12 +31,15 @@ public function string()
      */
     private function parameterComponents()
     {
-        return [
-            sprintf('keyId="%s"', $this->key->id),
-            sprintf('algorithm="%s"', $this->algorithm->name()),
-            sprintf('headers="%s"', $this->headerList->string()),
-            sprintf('signature="%s"', $this->signatureBase64()),
-        ];
+        $components = [];
+        $components[] = sprintf('keyId="%s"', $this->key->id);
+        $components[] = sprintf('algorithm="%s"', $this->algorithm->name());
+        if ($this->headerList->headerListSpecified()) {
+            $components[] = sprintf('headers="%s"', $this->headerList->string());
+        }
+        $components[] = sprintf('signature="%s"', $this->signatureBase64());
+
+        return $components;
```

Comment:
this change needs a corresponding test

Reply:
No Headers Signing Context: https://github.com/liamdennehy/http-signatures-php/blob/e8d9b91904f43f80af78bca6cacbc72208669a06/tests/ContextTest.php#L24
No Headers Signing Test: https://github.com/liamdennehy/http-signatures-php/blob/e8d9b91904f43f80af78bca6cacbc72208669a06/tests/ContextTest.php#L195
No Headers Verifier Message: https://github.com/liamdennehy/http-signatures-php/blob/e8d9b91904f43f80af78bca6cacbc72208669a06/tests/VerifierTest.php#L58
No Header IsValid (correctly signed) test: https://github.com/liamdennehy/http-signatures-php/blob/e8d9b91904f43f80af78bca6cacbc72208669a06/tests/VerifierTest.php#L79
All other variants (e.g. signing wrong headers) already covered, more required?

Reply:
Closing...